src/app/marketplace/edit-ad.component.ts

```diff
@@ -47,6 +48,10 @@ export class EditAdComponent
 
   images$ = new BehaviorSubject<Image[]>([]);
   currency$ = new BehaviorSubject<Currency>(null);
+  decimalQuantity$ = new BehaviorSubject<number>(null);
```

Comment:
It is not needed to mirror the form fields in `BehaviorSubject`s, only if the type is different (ex: Currency vs string).
you can use the async pipe like `form.get('decimalQuantity').valueChanges | async` directly.